util/settings/settings.go

```diff
@@ -323,6 +325,8 @@ const (
 	partOfArgoCDSelector = "app.kubernetes.io/part-of=argocd"
 	// settingsPasswordPatternKey is the key to configure user password regular expression
 	settingsPasswordPatternKey = "passwordPattern"
+	// inClusterServerAddressAllowedKey is the key to configure whether to allow in-cluster server address
+	inClusterServerAddressAllowedKey = "inClusterServerAddressAllowed"
```

Comment:
Let's nest this under a `cluster` settings namespace (like we do with repository, application, resource, etc...):

```yaml
data:
  cluster.inClusterEnabled: false
```